core/node/node_framework/src/implementations/layers/eth_watch.rs

```diff
@@ -114,40 +116,53 @@ impl WiringLayer for EthWatchLayer {
             self.chain_id,
         );
 
-        let sl_l2_client: Box<dyn ZkSyncExtentionEthClient> = Box::new(EthHttpQueryClient::new(
-            input.gateway_client.0,
-            input
-                .contracts_resource
-                .0
-                .chain_contracts_config
-                .diamond_proxy_addr,
-            input
-                .l1ecosystem_contracts_resource
-                .0
-                .bytecodes_supplier_addr,
-            input
-                .l1ecosystem_contracts_resource
-                .0
-                .wrapped_base_token_store,
-            input.l1ecosystem_contracts_resource.0.shared_bridge,
-            input
-                .contracts_resource
-                .0
-                .ecosystem_contracts
-                .state_transition_proxy_addr,
-            input
-                .contracts_resource
-                .0
-                .chain_contracts_config
-                .diamond_proxy_addr,
-            input
-                .contracts_resource
-                .0
-                .ecosystem_contracts
-                .server_notifier_addr,
-            self.eth_watch_config.confirmations_for_eth_event,
-            self.chain_id,
-        ));
+        macro_rules! create_client {
```

Comment:
This is unnecessary. You can extract a function

```rust
impl EthWatchLayer {
    fn create_client<Net: Network>(
        &self,
        client: Box<DynClient<Net>>,
        l1_contracts: &L1ChainContractsResource,
        l1ecosystem_contracts_resource: &L1EcosystemContractsResource,
    ) -> EthHttpQueryClient<Net> {
        // ...
    }
}
```

...and use it instead of this macro and above to create `l1_client`.